### PR TITLE
participant_contracts write path

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Writes contracts alongside events for interpretation and validation.

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/resources/db/migration/h2database/V18__Contracts_new_schema.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+-- visibility of contracts to parties
+create table participant_contract_witnesses
+(
+    contract_id varchar not null,
+    contract_witness varchar not null,
+
+    foreign key (contract_id) references participant_contract_witnesses(contract_id)
+);
+create index on participant_contract_witnesses(contract_id);      -- join with contracts
+create index on participant_contract_witnesses(contract_witness); -- filter by party
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/main/resources/db/migration/h2database/V18__Contracts_new_schema.sql:34]: Newline plz.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/EventsTableInsert.scala [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 package com.daml.platform.store.dao.events
 
-import java.util.Date
+import java.time.Instant
```
Comment (SamirTalwar-DA) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/EventsTableInsert.scala:6]: 🥳

File: ledger/sandbox/src/main/resources/db/migration/h2database/V18__Contracts_new_schema.sql (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/resources/db/migration/h2database/V18__Contracts_new_schema.sql:36]: Addressed by https://github.com/digital-asset/daml/pull/5522/commits/689a634c59affe120943c11d76b99a14e62a8685

**Comment by stefanobaghino-da** [comment]
I noticed that I was creating the disclosure batch even if no contract was being persisted, fixed in https://github.com/digital-asset/daml/pull/5522/commits/a39ddca33b068335a64beaec8008ce187211da43.

**Comment by stefanobaghino-da** [comment]
More importantly, attempting to add witnesses for contracts that are not persisted (because they're transient) caused a key constraint error. Fixed in https://github.com/digital-asset/daml/pull/5522/commits/81fe49eb6caaf43076edcafc91a96b4370c347ee.

**Comment by stefanobaghino-da** [comment]
And of course I forgot to drop the new tables on reset, fixed in https://github.com/digital-asset/daml/pull/5522/commits/a361a35861d826990e743389ef602bc93f05f6e2.

**Pull request merged**